### PR TITLE
[CONTP-411] exclude syncing pod tags in clc tagger

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

This PR updates the protobuf of the tagger stream request so that the client can communicate the prefixes of the entities that it needs to sync into the remote tagger. 

It also updates the CLC runner remote tagger so that pod tags are excluded from the sync.

### Motivation

Stop syncing pod tags from the cluster tagger to the clc runner remote tagger. 
The target is to reduce the network, memory and cpu load, and thereby allow clc runners to scale up smoothly on large clusters.

### Additional Notes

The cluster check runners now miss pod tags in the tagger when using the cluster tagger, but we already have [this](https://github.com/DataDog/datadog-agent/pull/29209) PR which allows CLC runners to calculate pod tags on the fly when needed without having to sync pod tags in the tagger.

### Possible Drawbacks / Trade-offs

Further optimisation:

The tagger server handling of streams initial sync can be optimised by maintaining a pool with a max number of connections that can run initial sync at the same time in parallel.

For now, cutting off pod tags from the sync should itself be a huge gain on large clusters. We will work on optimising this more if we find this necessary.

### Describe how to test/QA your changes

E2E tests and unit tests should be enough to handle testing. 

But as a double verification, it is recommended to verify that CLC runners can now sync with cluster tagger while excluding pod tags. 

Here are the steps:

1. Deploy the agent and cluster agent with the orchestrator cluster check running in a CLC runner and using the cluster tagger. Also configure namespace labels as tags and enable kubernetes tags collection so that the cluster tagger has some tags for pods and namespaces:

```
datadog:
  apiKeyExistingSecret: datadog-secret
  appKeyExistingSecret: datadog-secret
  clusterChecks:
    enabled: true
  kubelet:
    tlsVerify: false

  clusterTagger:
    collectKubernetesTags: true

  namespaceLabelsAsTags:
    kubernetes.io/metadata.name: name

clusterChecksRunner:
  enabled: true
  replicas: 1
  env:
    - name: DD_CLC_RUNNER_REMOTE_TAGGER_ENABLED
      value: true

clusterAgent:
  enabled: true
  replicas: 1

  advancedConfd:
    orchestrator.d:
      1.yaml: |-
        cluster_check: true
        init_config:
        instances:
          - collectors:
            - pods
            skip_leader_election: true

```

2. Check that the cluster tagger contains tags for pods and namespaces:

```
kubectl exec <cluster-agent-pod-name> -- agent tagger-list

=== Entity kubernetes_metadata:///namespaces//default ===
== Source workloadmeta-kubernetes_metadata =
=Tags: [name:default]
===

=== Entity kubernetes_metadata:///namespaces//kube-public ===
== Source workloadmeta-kubernetes_metadata =
=Tags: [name:kube-public]
===

=== Entity kubernetes_pod_uid://481c1f63-89ef-4f7e-8b92-25b9e4df2add ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_app_component:clusterchecks-agent kube_app_instance:datadog-agent kube_app_managed_by:Helm kube_app_name:datadog-agent kube_cluster_name:adel-orchestrator-clc kube_deployment:datadog-agent-clusterchecks kube_namespace:default kube_ownerref_kind:replicaset kube_ownerref_name:datadog-agent-clusterchecks-574dfbd86f kube_qos:BestEffort kube_replica_set:datadog-agent-clusterchecks-574dfbd86f pod_name:datadog-agent-clusterchecks-574dfbd86f-ptp8t pod_phase:running]
===

=== Entity kubernetes_metadata:///namespaces//kube-system ===
== Source workloadmeta-kubernetes_metadata =
=Tags: [name:kube-system]
===

=== Entity kubernetes_metadata:///namespaces//local-path-storage ===
== Source workloadmeta-kubernetes_metadata =
=Tags: [name:local-path-storage]
===

=== Entity kubernetes_pod_uid://73d766d6-869e-4e89-8a15-63c11eb6ffb3 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_namespace:kube-system kube_ownerref_kind:node kube_ownerref_name:kind-control-plane kube_priority_class:system-node-critical kube_qos:Burstable pod_name:kube-apiserver-kind-control-plane pod_phase:running]
===

=== Entity kubernetes_pod_uid://75491183-12b9-483f-a8bd-d573f983a53b ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_namespace:default kube_qos:BestEffort pod_name:unschedulable-pod pod_phase:pending]
===

=== Entity kubernetes_pod_uid://ebb6d4f5-9850-4b4a-9a4c-5c7418b2f912 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_namespace:kube-system kube_ownerref_kind:node kube_ownerref_name:kind-control-plane kube_priority_class:system-node-critical kube_qos:Burstable pod_name:kube-scheduler-kind-control-plane pod_phase:running]
===

=== Entity kubernetes_metadata:///namespaces//kube-node-lease ===
== Source workloadmeta-kubernetes_metadata =
=Tags: [name:kube-node-lease]
===

=== Entity kubernetes_pod_uid://3956c4f6-32df-4f6a-b5be-72149745bb00 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_namespace:kube-system kube_ownerref_kind:node kube_ownerref_name:kind-control-plane kube_priority_class:system-node-critical kube_qos:Burstable pod_name:kube-controller-manager-kind-control-plane pod_phase:running]
===

=== Entity kubernetes_pod_uid://541a6a3e-f829-434c-9bb7-dcf3b8840bb2 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_daemon_set:kube-proxy kube_namespace:kube-system kube_ownerref_kind:daemonset kube_ownerref_name:kube-proxy kube_priority_class:system-node-critical kube_qos:BestEffort pod_name:kube-proxy-d2bj9 pod_phase:running]
===

=== Entity kubernetes_pod_uid://620e8961-83ea-4f3d-9256-0bb67b2b959d ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_deployment:local-path-provisioner kube_namespace:local-path-storage kube_ownerref_kind:replicaset kube_ownerref_name:local-path-provisioner-6bc4bddd6b kube_qos:BestEffort kube_replica_set:local-path-provisioner-6bc4bddd6b pod_name:local-path-provisioner-6bc4bddd6b-qjrtf pod_phase:running]
===

=== Entity kubernetes_pod_uid://8ef52246-fd65-48ec-b04d-afc5ab949c25 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_daemon_set:kindnet kube_namespace:kube-system kube_ownerref_kind:daemonset kube_ownerref_name:kindnet kube_qos:Guaranteed pod_name:kindnet-4lzgr pod_phase:running]
===

=== Entity kubernetes_pod_uid://fce5e4a8-db5e-4b4b-8cab-be6029e1b762 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_namespace:kube-system kube_ownerref_kind:node kube_ownerref_name:kind-control-plane kube_priority_class:system-node-critical kube_qos:Burstable pod_name:etcd-kind-control-plane pod_phase:running]
===

=== Entity internal://global-entity-id ===
== Source workloadmeta-static =
=Tags: [kube_cluster_name:adel-orchestrator-clc]
===

=== Entity kubernetes_pod_uid://184c26b0-f03b-4887-896c-092250f61138 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_app_component:cluster-agent kube_app_instance:datadog-agent kube_app_managed_by:Helm kube_app_name:datadog-agent kube_cluster_name:adel-orchestrator-clc kube_deployment:datadog-agent-cluster-agent kube_namespace:default kube_ownerref_kind:replicaset kube_ownerref_name:datadog-agent-cluster-agent-78cdcb7c8 kube_qos:BestEffort kube_replica_set:datadog-agent-cluster-agent-78cdcb7c8 pod_name:datadog-agent-cluster-agent-78cdcb7c8-sf9l4 pod_phase:running]
===

=== Entity kubernetes_pod_uid://23b009dc-8790-41df-a6ad-d315ec695b7b ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_deployment:coredns kube_namespace:kube-system kube_ownerref_kind:replicaset kube_ownerref_name:coredns-5d78c9869d kube_priority_class:system-cluster-critical kube_qos:Burstable kube_replica_set:coredns-5d78c9869d pod_name:coredns-5d78c9869d-7tsr5 pod_phase:running]
===

=== Entity kubernetes_pod_uid://b77e5126-07bd-4c37-b613-616fb8ba38c1 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_app_component:clusterchecks-agent kube_app_instance:datadog-agent kube_app_managed_by:Helm kube_app_name:datadog-agent kube_cluster_name:adel-orchestrator-clc kube_deployment:datadog-agent-clusterchecks kube_namespace:default kube_ownerref_kind:replicaset kube_ownerref_name:datadog-agent-clusterchecks-574dfbd86f kube_qos:BestEffort kube_replica_set:datadog-agent-clusterchecks-574dfbd86f pod_name:datadog-agent-clusterchecks-574dfbd86f-fwkw5 pod_phase:running]
===

=== Entity kubernetes_pod_uid://c4c80235-9a96-4427-9b60-e146a9477eda ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_cluster_name:adel-orchestrator-clc kube_deployment:coredns kube_namespace:kube-system kube_ownerref_kind:replicaset kube_ownerref_name:coredns-5d78c9869d kube_priority_class:system-cluster-critical kube_qos:Burstable kube_replica_set:coredns-5d78c9869d pod_name:coredns-5d78c9869d-frvkc pod_phase:running]
===

=== Entity kubernetes_pod_uid://c4ec4af1-9b22-4273-98dd-6780ceb42563 ===
== Source workloadmeta-kubernetes_pod =
=Tags: [kube_app_component:agent kube_app_instance:datadog-agent kube_app_managed_by:Helm kube_app_name:datadog-agent kube_cluster_name:adel-orchestrator-clc kube_daemon_set:datadog-agent kube_namespace:default kube_ownerref_kind:daemonset kube_ownerref_name:datadog-agent kube_qos:BestEffort pod_name:datadog-agent-d5j2t pod_phase:running]
===

```

3. Verify that only namespace tags are synced into the CLC runner remote tagger:

```
kubectl <clc_runner_pod_name> -- agent tagger-list


=== Entity kubernetes_metadata:///namespaces//default ===
== Source remote =
=Tags: [name:default]
===

=== Entity kubernetes_metadata:///namespaces//kube-node-lease ===
== Source remote =
=Tags: [name:kube-node-lease]
===

=== Entity kubernetes_metadata:///namespaces//kube-public ===
== Source remote =
=Tags: [name:kube-public]
===

=== Entity kubernetes_metadata:///namespaces//kube-system ===
== Source remote =
=Tags: [name:kube-system]
===

=== Entity kubernetes_metadata:///namespaces//local-path-storage ===
== Source remote =
=Tags: [name:local-path-storage]
===

```

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by agent shared components.

**Review by mellon85 — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]
only changes for processes seem to be the autogenerated version bumps in `pkg/proto/pbgo/process/process.pb.go` and `pkg/proto/pbgo/process/workloadmeta_process.pb.go`

File: pkg/proto/pbgo/core/api.pb.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.34.0
-// 	protoc        v5.26.1
+// 	protoc-gen-go v1.28.1
```
[the remark this answers is not in the document]
Comment (adel121) [reply pkg/proto/pbgo/core/api.pb.go:3]: These files are generated by the `inv` command. So I am not sure if the behaviour is intended or not. 

I will rebase on main and check again.

**Review by davidor — APPROVED** [review]

**Pull request merged**